src/main/scala/mesosphere/marathon/core/pod/PodDefinition.scala

```diff
@@ -148,4 +149,20 @@ object PodDefinition {
   val DefaultBackoffStrategy = BackoffStrategy()
   val DefaultUpgradeStrategy = AppDefinition.DefaultUpgradeStrategy
   val DefaultUnreachableStrategy = UnreachableStrategy.default(resident = false)
+
+  private[core] def aggregateResourceLimits(resourceLimits: Seq[ResourceLimits]): Option[ResourceLimits] = {
+    val cpusLimit = resourceLimits.iterator.flatMap(_.cpus).sum
+    val memLimit = resourceLimits.iterator.flatMap(_.mem).sum
+    if ((cpusLimit == 0d) && (memLimit == 0d)) {
+      None
+    } else {
+      Some(
+        ResourceLimits(
+          cpus = (if (cpusLimit == 0d) None else Some(cpusLimit)),
+          mem = (if (memLimit == 0d) None else Some(memLimit))
+        )
+      )
+    }
```

Comment:
only now i realize that this won't help the frontend much.  
i was looking to have marathon factor in the number of currently running instances as well as the executor needs.

here are the current calculations: https://github.com/dcos/dcos-ui/blob/69d94db858d6c406c43846db586b57ca09100de1/plugins/services/src/js/columns/resourceLimits.ts#L34

i also think that we should make a difference between `None` and `0` as we recently encountered quota issues due to exactly not making that difference: https://jira.d2iq.com/browse/COPS-6589.